### PR TITLE
ENH: Update to flake8 3.2.1

### PR DESCRIPTION
**Pull request by dopplershift**
This fixes a problem with plugins not reporting errors properly: https://gitlab.com/pycqa/flake8/issues/257

### CODE REVIEW

**Comment by dopplershift** [comment]
Ok, so the useless error message `AttributeError` is caused by a bug in flake8 after handling an `IOError`. So the real question is: why would it hit an `IOError` trying to run `flake8` on stuff in `${SRC_DIR}`? Any why is it a problem now?

**Comment by dopplershift** [comment]
Out of curiosity, why run flake8 against the source files? Why not just run `flake8 --version` and be done with it? I mean, we're just testing that the package isn't horribly borked (meaning we can run the script and it finds the install of the library).

Any thoughts guys?

**Comment by dopplershift** [comment]
I see, that test was added to make sure we had a well-functioning set of dependencies. So, what could cause it to bork on windows...

**Comment by croth1** [comment]
@jakirkham any idea what happened here? Seems like some files do not exist in ${SRC_DIR} on windows, or do not have read permission. This is weird, because macOS and linux pass and windows passed a week ago as well. Maybe restarting will already help?

@dopplershift, do you have a windows machine to test if the recipe builds fine locally? As a workaround you could exclude the test on windows via selectors and test printing the version instead. Would be nice to understand what's going wrong here, though.

**Comment by dopplershift** [comment]
I agree--I've even tried rebuilding it multiple times to no avail.

I have a VM I can give a try here in a few minutes. Failing that, a selector is probably a decent way to go. (If `flake8` didn't have that `IOError` bug, we could at least understand what the failure **is**.)

**Comment by dopplershift** [comment]
Ok, I'm able to reproduce locally--**only** with the 3.2.1 source distribution, so I'm guessing there's a funky file. I think I'm going to add a patch to fix that `AttributeError` so I can figure out WTF is going on.

**Comment by dopplershift** [comment]
Well, here's the problem now that I've patched it:
```
${SRC_DIR}/setup.py:0:1: E902 IOError: [Errno 2] No such file or directory: '${SRC_DIR}/setup.py'
${SRC_DIR}/src/flake8/:0:1: E902 IOError: [Errno 2] No such file or directory: '${SRC_DIR}/src/flake8/'
${SRC_DIR}/tests:0:1: E902 IOError: [Errno 2] No such file or directory: '${SRC_DIR}/tests'
```

**Comment by dopplershift** [comment]
AHA! So the problem is that ${SRC_DIR} isn't getting interpolated properly on windows. The difference between 3.2.0 and 3.2.1 is that the non-existent directories were silently skipped in 3.2.0 (you can see this by running with `-v -v`). So these tests *never executed* on any files on windows before.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,7 +35,8 @@ test:
   imports:
     - {{ name }}
   commands:
-    - flake8 ${SRC_DIR}/setup.py ${SRC_DIR}/src/flake8/ ${SRC_DIR}/tests
+    - flake8 ${SRC_DIR}/setup.py ${SRC_DIR}/src/flake8 ${SRC_DIR}/tests  # [unix]
+    - flake8 %SRC_DIR%\\setup.py %SRC_DIR%\\src\\flake8 %SRC_DIR%\\tests  # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: 👍

Another viable option would be to change `${SRC_DIR}` to `{{SRC_DIR}]`. That would use jinja and work on both platforms. Though there may be some tweaks for the linter.

**Comment by jakirkham** [comment]
> Seems like some files do not exist in ${SRC_DIR} on windows, or do not have read permission.

As noted, it was an oversight that `${SRC_DIR}` is use at all on Windows. Should be good with @dopplershift 's change. 👍

**Pull request merged**